ledger/ledger-runner-common/src/test/scala/com/daml/ledger/runner/common/PureConfigReaderWriterSpec.scala

```diff
@@ -0,0 +1,743 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.runner.common
+
+import com.daml.lf.interpretation.Limits
+import com.daml.lf.language.LanguageVersion
+import com.daml.lf.transaction.ContractKeyUniquenessMode
+import com.daml.lf.{VersionRange, language}
+import com.typesafe.config.ConfigValueFactory.fromAnyRef
+import org.scalacheck.Gen
+import org.scalatest.{Assertion, EitherValues}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import pureconfig.{ConfigReader, ConfigSource}
+import PureConfigReaderWriter._
+import com.daml.ledger.api.tls.{SecretsUrl, TlsVersion}
+import com.daml.metrics.MetricsReporter
+import com.daml.platform.apiserver.{ApiServerConfig, AuthServiceConfig}
+import com.daml.platform.apiserver.SeedService.Seeding
+import com.daml.platform.apiserver.configuration.RateLimitingConfig
+import com.daml.platform.configuration.{
+  CommandConfiguration,
+  IndexServiceConfig,
+  InitialLedgerConfiguration,
+  PartyConfiguration,
+}
+import com.daml.platform.indexer.IndexerConfig
+import com.daml.platform.indexer.ha.HaConfig
+import com.daml.platform.services.time.TimeProviderType
+import com.daml.platform.store.DbSupport.ParticipantDataSourceConfig
+import com.daml.platform.store.backend.postgresql.PostgresDataSourceConfig.SynchronousCommitValue
+import com.daml.platform.usermanagement.UserManagementConfig
+import com.typesafe.config.ConfigFactory
+import pureconfig.error.ConfigReaderFailures
+
+import java.net.InetSocketAddress
+import java.nio.file.Path
+import java.time.Duration
+
+class PureConfigReaderWriterSpec
+    extends AnyFlatSpec
+    with Matchers
+    with ScalaCheckPropertyChecks
+    with EitherValues {
+
+  def convert[T](converter: ConfigReader[T], str: String): Either[ConfigReaderFailures, T] = {
+    val value = ConfigFactory.parseString(str)
+    for {
+      source <- ConfigSource.fromConfig(value).cursor()
+      result <- converter.from(source)
+    } yield result
+  }
+
+  behavior of "Duration"
+
+  it should "be isomorphic" in forAll(ArbitraryConfig.duration) { generatedValue =>
+    val configValue = javaDurationWriter.to(generatedValue)
+    javaDurationReader.from(configValue).value shouldBe generatedValue
+  }
```

Comment:
You have a lot of tests for isomorphism. Can you DRY them up? 
e.g. 
```suggestion
  it should "be isomorphic" in forAll(ArbitraryConfig.duration)(
    testIsomorphic(javaDurationWriter, javaDurationReader)
  )

  private def testIsomorphic[A](writer: ConfigWriter[A], reader: ConfigReader[A])(
      generatedValue: A
  ): Assertion = {
    val configValue = writer.to(generatedValue)
    reader.from(configValue).value shouldBe generatedValue
  }
```
This should save some tens of LOC